### ISSUE TITLE
static_acquire_release_test fails to build on CMake Xcode build

### ISSUE THREAD

**Issue by abestephensg** [halide/Halide#582]
Several of the generator based tests fail to build on Mac OS X / Xcode due to not finding the script generate_acquire_release.

Steps to reproduce:
1.) Generate Halide.xcodeproj using ccmake -GXcode
2.) Open the project in Xcode and build the "static_acquire_release_test" target

cd /Users/abstephens/Halide-master/build-xcode/test/scratch_generate_acquire_release && /Users/abstephens/Halide-master/build-xcode/bin/generate_acquire_release
/bin/sh: /Users/abstephens/Halide-master/build-xcode/bin/generate_acquire_release: No such file or directory
make: **\* [/Users/abstephens/Halide-master/build-xcode/test/libacquire_release.a] Error 127

**Comment by fish2000** [comment]
I am also getting an error when building the `acquire_release` generator test target – also on OS X, but I run into the problem when building libHalide as a dylib using Makefiles.

The verbose Makefile output surrounding the error: https://gist.github.com/fish2000/2ce4a87c7b0f2fd5b4d0

… Oddly, if I run the offending individual command(s) manually after the error cmake’s Makefile halts, I get no errors, and it *looks* like the command executes OK (files are created) but there is no output beyond that to report from these manual invocations and I can’t be sure that they work, really – only that they do something different that exits with status 0.

If I patch `CMakeLists.txt` to skip the tests, the build works; thereafter I can successfully build and run all non-generator-related tests. (I use this Homebrew formula to do the _sans_-test build patch: https://github.com/fish2000/homebrew-praxa/blob/master/halide.rb)

Thanks!

**Comment by steven-johnson** [comment]
Is this happening as of this morning's Halide tip? (I was seeing this
yesterday afternoon but various pushes late yesterday addressed this for
me.)

**Comment by fish2000** [comment]
I had the same issue when I built with the latest from master circa yesterday afternoon – let me know if that is the proper branch and/or what I might cherry-pick for the fix, and I will retry this evening and report back. Thanks for the reply!

**Comment by steven-johnson** [comment]
Try syncing to (at least)
https://github.com/halide/Halide/commit/4c9242993ab119f0d2b598733c47274ea6dcbd52

**Comment by fish2000** [comment]
@steven-johnson – Alas, I have tried again thoroughly and I got the same result even with all sorts of permutations:

* With and without homebrew;
* Rebuilding all of LLVM/LLDB/Clang (version 3.51) in between, building in the broadest-possible spectrum of backends just to be sure;
* With both Git HEAD and with a clone from the commit [you referenced above](https://github.com/halide/Halide/issues/582#issuecomment-89040455).

… The only thing that works for sure, on my end, is not building any tests. 

[Like I said earlier](https://github.com/halide/Halide/issues/582#issuecomment-88985946), I can still issue somewhat successful (albeit specious) manual build commands for generator tests, so I will try all that and if I get anything of forensic noteworthiness I will post it. 

Thanks again sir!
